### PR TITLE
[W5.3b][W13-A1] Ang Jing Zhe

### PR DESCRIPTION
**Pull request by AJZ1995**
add inheritance to contact

### CODE REVIEW

**Comment by AJZ1995** [comment]
I edited the code and have no idea how to update it.

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Input validation purpose
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:4]: Incorrect / irrelevant header comment.
Please understand the purpose of header comments: other developers who extend your work should get necessary information about your classes and their methods.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+public class Contact {
+
+    public final String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:9]: Try to make it private. 
In general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public boolean isPrivate;
+
+    public Contact(String _value, boolean _isPrivate) {
+        value = _value;
+        isPrivate = _isPrivate;
+    }
+
+    @Override
+    public int hashCode()
+    {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:19]: Coding standard violation

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public final String value;
+    public boolean isPrivate;
+
+    public Contact(String _value, boolean _isPrivate) {
+        value = _value;
+        isPrivate = _isPrivate;
+    }
+
+    @Override
+    public int hashCode()
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java:18]: In general, `equals()` and `hashCode()` methods should be overridden together. Find out why!

**Review by okkhoy — COMMENTED** [review]
You can try to move the validation logic as well to the superclass.

**Comment by okkhoy** [comment]
@AJZ1995 
Some comments added. Please ack & close the PR after reading comments.

**Comment by AJZ1995** [comment]
Ack

**Pull request closed without merging**